### PR TITLE
ENH: Add info SSS basis

### PR DESCRIPTION
**Pull request by Eric89GXL**
@kambysese can you have a look?

@mdclarke @staulu you should be able to do this:
```
import numpy as np
from scipy import linalg

import mne
from mnefun import info_sss_basis

data_path = mne.datasets.sample.data_path()
evoked = mne.read_evokeds(data_path + '/MEG/sample/sample_audvis-ave.fif')[0]
evoked = evoked.pick_types(meg=True)  # only look at MEG channels
S = info_sss_basis(evoked.info)
sing = linalg.svdvals(S)
print('Condition before proj:  %s' % (sing[0] / sing[-1],))
proj_op = mne.io.proj.make_projector_info(evoked.info)[0]
S_proj = np.dot(proj_op, S)
sing = linalg.svdvals(S_proj)
print('Condition after proj:   %s' % (sing[0] / sing[-1],))
```
And see:
```
Condition before proj:  59.0775421459
Condition after proj:   135.880057672
```

### CODE REVIEW

**Comment by mdclarke** [comment]
I keep getting the error :
ImportError: cannot import name info_sss_basis

even though I am up to date with current LABSN/mnefun

-Maggie

**Comment by Eric89GXL** [comment]
You need to be on the branch in my fork that forms this PR, not `master`. Let me know if you can't figure out how to do it

File: mnefun/_mnefun.py [hunk 1]
```diff
@@ -980,12 +991,11 @@ def run_sss_locally(p, subjects, run_indices):
...
-                                         cross_talk=ct_file,
-                                         st_correlation=p.st_correlation, st_duration=st_duration,
-                                         destination=None,
-                                         coord_frame='meg')
+                raw_sss = maxwell_filter(
+                    raw, int_order=p.int_order, ext_order=p.ext_order,
+                    calibration=cal_file, cross_talk=ct_file,
+                    st_correlation=p.st_correlation, st_duration=st_duration,
+                    destination=None, coord_frame='meg')
                 raw_sss.save(o, overwrite=True, buffer_size_sec=None)
```
Comment (kambysese) [inline mnefun/_mnefun.py]: Do you have some preferred line length I should know about? :)

File: mnefun/_mnefun.py [hunk 2]
```diff
@@ -2123,4 +2135,72 @@ def _headpos(p, file_in, file_out):
...
+                         % coord_frame)
+    origin = _check_origin(origin, info, 'head')
+    regularize = _check_regularize(regularize, ('in', 'svd'))
+    meg_picks, mag_picks, grad_picks, good_picks, coil_scale, mag_or_fine = \
+        _get_mf_picks(info, int_order, ext_order, ignore_ref)
+    info_from = pick_info(info, good_picks, copy=True)
+    all_coils_recon = _prep_mf_coils(info, ignore_ref=ignore_ref)
+    all_coils = _prep_mf_coils(info_from, ignore_ref=ignore_ref)
+    # remove MEG bads in "to" info
+    n_in, n_out = _get_n_moments([int_order, ext_order])
```
Comment (kambysese) [inline mnefun/_mnefun.py]: Should this be modifying `exp`? `n_in` & `n_out` are not used and I don't see how bad channels are removed in the "to" `info`

**Comment by kambysese** [comment]
Aside from minor comment LGTM. 
Just so that I am understanding this; the ratio between largest & smallest singular value goes from c. 60 to 135 after the projection is applied, meaning what? 
cc @staulu

**Comment by staulu** [comment]
It means that our ability to reconstruct all brain signals deteriorates due
to the fact that we modify the brain-related signal subspace by the
orthogonal projection. I expected the ratio of the largest & smallest
singular value (the condition number of a matrix) to increase due to the
projection, and 60 -> 135 is not worrisome. We are far from singularity and
the reconstruction operators to compensate for the projection-related
signal space rotations should work fine.

As the aforementioned ratio increases, first the source level estimates
tend to get noisier (in terms of random measurement noise), and finally as
the ratio approaches infinity we lose our ability to reconstruct some of
the underlying brain signal altogether.

I'd be curious to see what kind of numbers you get from different cases

Samu


Samu Taulu, D.Sc.
Research Associate Professor, Physics
Director, I-LABS MEG Brain Imaging Center
Institute for Learning & Brain Sciences (I-LABS)
Portage Bay Bldg. Room 212
Box 357988
University of Washington
Seattle, WA 98195-7988
USA

File: mnefun/_mnefun.py (hunk 1 above)
Comment (Eric89GXL) [inline mnefun/_mnefun.py]: Yeah, PEP8 (80 char). I thought you had PEP8 checking enabled?

Comment (kambysese) [inline mnefun/_mnefun.py]: Yeah, PEP8 checking is in effect and I am not getting any notifications that lines are running long. I have noticed that between pycharm, atom, and spyder editors the character demarcation results in shorter lines in e.g. atom vs. pycharm. I'll look into what pycharm is doing.

**Comment by Eric89GXL** [comment]
Thanks for the comments @kambysese, addressed and merged

**Pull request merged**